libs/live-countervalues-react/src/index.tsx

```diff
@@ -90,6 +151,15 @@ export function useTrackingPairForAccounts(
   return useMemo(() => c.pairs, [c.hash]);
 }
 
+export function CountervaluesMarketcap({ children }: { children: React.ReactNode }): ReactElement {
+  const marketcapIds = useMarketcap();
```

Comment:
As it is a hook, why just not use it when it's needed instead of doing a Context that will do a lot of rerender when ids are refetched?

I'm maybe wrong

Reply:
@mcayuelas-ledger I tried to cover this part in

```
<CountervaluesMarketcap> context is introduced to be able to share the same cache of fetching the available currencies of the API and sorted by marketcap. It is not required to depends on it, but it is strongly recommended to migrate to it. the older way to do currenciesByMarketcap is still functional but deprecated in favor of this new paradigm that we will progressively migrate to.
```

Here is what I understand of how to use React Context so far:

it's better to re-render a context once, than having N parts of the app that independently load again the data.

in theory, what you explained should not happen if we properly memoize components. here is an example: let's say we have this tree of component:

A -> B -> C.

A renders B which renders C (from parent to children).

when we make this 

```
<CountervaluesMarketcap>
  <A />
  </CountervaluesMarketcap>
```

as soon as A is properly memo ized, it will not re-render.

and when C will need to `useContextMarketcap()` (which is the context version of the hook, hook that share the same state of marketcap) it will indeed re-render, like if it was a direct usage of the useMarketcap() one. but A and B will not re-render in that case.

normally our app is using react router which itself will memoize the pages. we probably will want to double confirm that it's the case, because it will benefit all our existing contexts usage.

Reply:
Indeed :) 
Sounds good !